src/analysis/normed_space/weak_dual.lean

```diff
@@ -137,4 +137,52 @@ end
 
 end normed_space.dual
 
+namespace weak_dual
+
+lemma to_normed_dual.preimage_closed_unit_ball :
+  (to_normed_dual ⁻¹' metric.closed_ball (0 : dual 𝕜 E) 1) =
+    {x' : weak_dual 𝕜 E | ∥ x'.to_normed_dual ∥ ≤ 1} :=
+begin
+  have eq : metric.closed_ball (0 : dual 𝕜 E) 1 = {x' : dual 𝕜 E | ∥ x' ∥ ≤ 1},
+  { ext x', simp only [dist_zero_right, metric.mem_closed_ball, set.mem_set_of_eq], },
+  rw eq,
+  exact set.preimage_set_of_eq,
+end
+
+variables (𝕜)
+
+/-- The polar set `polar 𝕜 s` of `s : set E` seen as a subset of the dual of `E` with the
+weak-star topology is `weak_dual.polar 𝕜 s`. -/
+def polar (s : set E) : set (weak_dual 𝕜 E) := to_normed_dual ⁻¹' (polar 𝕜 s)
+
+end weak_dual
+
 end weak_star_topology_for_duals_of_normed_spaces
+
+section polar_sets_in_weak_dual
+
+open metric set normed_space
+
+namespace polar
+
+variables {𝕜 : Type*} [nondiscrete_normed_field 𝕜]
+variables {E : Type*} [normed_group E] [normed_space 𝕜 E]
+
+/-- The polar `polar 𝕜 s` of a set `s : E` is a closed subset when the weak star topology
+is used, i.e., when `polar 𝕜 s` is interpreted as a subset of `weak_dual 𝕜 E`. -/
+lemma is_weak_dual_closed (s : set E) : is_closed (weak_dual.polar 𝕜 s) :=
```

Comment:
I will never guess that name. What about `weak_dual.is_closed_polar`?

Reply:
Thanks for the good suggestion! Changed.